### DISCUSSION TITLE
Error when training

### DISCUSSION THREAD

**Discussion by Paasu** [neuropoly/axondeepseg#538] [Help needed] [answered]
Using latest version of ADS and on OpenSUSE 15.1
Here is the training file and section of the log spat out when trying to train
[Files.zip](https://github.com/neuropoly/axondeepseg/files/6771633/Files.zip)

I ran the training.py and arrive with this error.

**Comment by jcohenadad** [comment, marked as answer]
Hi,
It says you got an OOM (out of memory), so it seems like the problem is not on ADS's side, but rather that the size of images is too much for the hardware you are using.

**Reply by Paasu** [reply]
I see that's odd I've been using these images before and had no issues. Would switching to 256x256 solve the issue?

**Reply by jcohenadad** [reply]
It is not the only parameter that affects RAM, but it could help. The bottomline is that you need to adjust the training parameters based on your hardware constraints, which are not infinite:
~~~
Limit:                  7879960167
InUse:                  7745749248
MaxInUse:               7745750016
NumAllocs:                    1749
MaxAllocSize:           3355443200
~~~

**Reply by Paasu** [reply]
I see, I have 16gb RAM and 8gb VRAM on my graphics card so what should the adjusted values be?

**Reply by vasudev-sharma** [reply]
Hey @Paasu you could try lowering your `batch_size` in the `config` file. OOM usaully occours when there memory limit the GPU is exhausted. You perhaps have a `batch_size` of **8** which you were using earlier wouldn't work on your GPU. 

A `batch_size` of **4** or even lower than that should probably work for your case. 

Another solution is to use small **patch size**, which you suggested earlier **(256 x 256)** or even **(128 X 128)**. 

Could you first try playing with **batch size** and if it doesn't work playing with **patch size**?

**Reply by Paasu** [reply]
Thank you! Lowering the batch-size to 6 has allowed the training to begin.
